include/igl/quadprog.h

```diff
@@ -0,0 +1,87 @@
+#ifndef IGL_QUADPROG_H
+#define IGL_QUADPROG_H
+#include "igl_inline.h"
+#include <Eigen/Core>
+namespace igl
+{
+  // Solve a convex quadratic program. Optimized for small dense problems.
+  // Still works for Eigen::Dynamic (and then everything needs to be Dynamic).
```

Comment:
So what are the differences between this one and `igl::copyleft::quadprog()`? Can we drop the latter now? How does it compare to [qpmad](https://github.com/asherikov/qpmad)?

Reply:
`igl::quadprog` implements a _very_ simple primal active set method. The new `igl::min_quad_with_fixed` is very fast for small dense problems so the iterations of `igl::quadprog` become very fast. Even _if_ it ends up doing many more iterations than `igl::copyleft::quadprog` it would be much faster (in reality it doesn't do _that_ many more iterations). It's a health 10-100x faster than `igl::copyleft::quadprog` for specific cases of QPs.

Unfortunately, that set is limited. `igl::quadprog` is really good at tiny box-constrained QPs with a positive definite objective (like the kind that show up in dual contouring). `igl::copyleft::quadprog` handles more general problems (and also starts to beat `igl::quadprog` when the number of variables gets over ~20). I tried extending `igl::quadprog` so that we could use it for `igl::copyleft::progressive_hulls` and drop `igl::copyleft::quadprog` but it was trickier than I thought. Something like `qpmad` or the non GPL version of `quadrog++` would be good future PR.

Reply:
Ok sounds good. It would be good to put your message in comments to the code so that it can serve as future reference.